src/tactic/norm_cast.lean

```diff
@@ -25,225 +25,314 @@ try_for 1000 (mk_instance e)
 end tactic
 
 namespace expr
+
 open tactic expr
 
 meta def flip_eq (ty : expr) : tactic (expr × (expr → expr)) :=
 do
-    (a, b) ← is_eq ty,
-    α ← infer_type a,
-    new_ty ← to_expr ``(%%b = %%a),
-    f ← to_expr ``(@eq.symm %%α %%a %%b),
-    return (new_ty, ⇑f)
+  (a, b) ← is_eq ty,
+  α ← infer_type a,
+  new_ty ← to_expr ``(%%b = %%a),
+  f ← to_expr ``(@eq.symm %%α %%a %%b),
+  return (new_ty, ⇑f)
 
 meta def flip_iff (ty : expr) : tactic (expr × (expr → expr)) :=
 do
-    (a, b) ← is_iff ty,
-    new_ty ← to_expr ``(%%b ↔ %%a),
-    f ← to_expr ``(@iff.symm %%a %%b),
-    return (new_ty, ⇑f)
+  (a, b) ← is_iff ty,
+  new_ty ← to_expr ``(%%b ↔ %%a),
+  f ← to_expr ``(@iff.symm %%a %%b),
+  return (new_ty, ⇑f)
+
+protected meta def to_pos_num : expr → option pos_num
```

Comment:
This, `to_num`, `pexpr_of_num`, `pexpr_of_pos_num` should be moved to `meta/expr.lean` near `to_pos_nat`, `to_nat`, etc.